zerver/models.py

```diff
@@ -874,61 +874,61 @@ def __str__(self) -> str:
+def realm_in_local_linkifiers_cache(realm_id: int) -> bool:
+    return realm_id in per_request_linkifiers_cache
 
 
-def realm_filters_for_realm(realm_id: int) -> List[Tuple[str, str, int]]:
-    if not realm_in_local_realm_filters_cache(realm_id):
-        per_request_realm_filters_cache[realm_id] = realm_filters_for_realm_remote_cache(realm_id)
-    return per_request_realm_filters_cache[realm_id]
+def linkifiers_for_realm(realm_id: int) -> List[Tuple[str, str, int]]:
+    if not realm_in_local_linkifiers_cache(realm_id):
+        per_request_linkifiers_cache[realm_id] = linkifiers_for_realm_remote_cache(realm_id)
+    return per_request_linkifiers_cache[realm_id]
 
 
-@cache_with_key(get_realm_filters_cache_key, timeout=3600 * 24 * 7)
-def realm_filters_for_realm_remote_cache(realm_id: int) -> List[Tuple[str, str, int]]:
+@cache_with_key(get_linkifiers_cache_key, timeout=3600 * 24 * 7)
+def linkifiers_for_realm_remote_cache(realm_id: int) -> List[Tuple[str, str, int]]:
     filters = []
-    for realm_filter in RealmFilter.objects.filter(realm_id=realm_id):
-        filters.append((realm_filter.pattern, realm_filter.url_format_string, realm_filter.id))
+    for linkifier in RealmFilter.objects.filter(realm_id=realm_id):
+        filters.append((linkifier.pattern, linkifier.url_format_string, linkifier.id))
 
     return filters
 
 
-def all_realm_filters() -> Dict[int, List[Tuple[str, str, int]]]:
+def linkifiers_from_all_realms() -> Dict[int, List[Tuple[str, str, int]]]:
     filters: DefaultDict[int, List[Tuple[str, str, int]]] = defaultdict(list)
-    for realm_filter in RealmFilter.objects.all():
-        filters[realm_filter.realm_id].append(
-            (realm_filter.pattern, realm_filter.url_format_string, realm_filter.id)
+    for linkifier in RealmFilter.objects.all():
+        filters[linkifier.realm_id].append(
+            (linkifier.pattern, linkifier.url_format_string, linkifier.id)
         )
 
     return filters
 
 
-def flush_realm_filter(sender: Any, **kwargs: Any) -> None:
+def flush_linkifiers(sender: Any, **kwargs: Any) -> None:
     realm_id = kwargs["instance"].realm_id
-    cache_delete(get_realm_filters_cache_key(realm_id))
+    cache_delete(get_linkifiers_cache_key(realm_id))
     try:
-        per_request_realm_filters_cache.pop(realm_id)
+        per_request_linkifiers_cache.pop(realm_id)
     except KeyError:
         pass
 
 
-post_save.connect(flush_realm_filter, sender=RealmFilter)
-post_delete.connect(flush_realm_filter, sender=RealmFilter)
+post_save.connect(flush_linkifiers, sender=RealmFilter)
+post_delete.connect(flush_linkifiers, sender=RealmFilter)
 
 
 def flush_per_request_caches() -> None:
     global per_request_display_recipient_cache
     per_request_display_recipient_cache = {}
-    global per_request_realm_filters_cache
-    per_request_realm_filters_cache = {}
+    global per_request_linkifiers_cache
+    per_request_linkifiers_cache = {}
```

Comment:
Definitely don't try to do this change in this commit, but I think we can actually rename `RealmFilter` to `Linkifier` in `models.py` with a small Django configuration option to set the database table name to still be `zerver_realmfilter`.

Reply:
I haven't put any of the following in this PR, but just writing my findings here...

There's a `db_table` meta property which we can specify for the table name. So a diff like this-

```diff
-class RealmFilter(models.Model):
+class Linkifier(models.Model):
     """Realm-specific regular expressions to automatically linkify certain
     strings inside the Markdown processor.  See "Custom filters" in the settings UI.
     """
@@ -869,6 +869,7 @@ class RealmFilter(models.Model):
     url_format_string: str = models.TextField(validators=[URLValidator(), filter_format_validator])

     class Meta:
+        db_table = "zerver_realmfilter"
         unique_together = ("realm", "pattern")
```
Results in a migration like so-
```python
    operations = [
        migrations.RenameModel(
            old_name='RealmFilter',
            new_name='Linkifier',
        ),
        migrations.AlterModelTable(
            name='linkifier',
            table='zerver_realmfilter',
        ),
    ]

```

Is that what you have in mind?

Another question I have is, why not change the database table name? I tried running migrations both ways -- specifying the `db_name` to be the old `zerver_realmfilter` and not specifying it (so that it defaults to `zerver_linkifier`). There seems to be no data loss in either case.